examples/llm_serving/launch_model_worker.py

```diff
@@ -79,22 +132,66 @@ def __init__(self,
             keys = json.load(open(KEYS_FILENAME, "r"))
             self.allowed_api_keys = keys["allowed_api_keys"]
 
+    def get_priority_queue(self) -> asyncio.PriorityQueue:
+        # Fetch the current priority queue
+        current_queue = self.wighted_queue_list[self.wrr_queue_idx]
+
+        # Prepare the index for the next iteration:
+        # The index remains the same unless we decide to also iterate over the priority queues.
+        # If so, implement an incremental iteration like the following:
+        # self.wrr_queue_idx = (self.wrr_queue_idx + 1) % len(self.wighted_queue_list)
+        # Decrement the total weight.
+        self.total_queue_weight -= 1
+        if self.total_queue_weight <= 0:
+            # If the total weight reaches 0, rotate the queue and start over
+            # with the next one, updating the total queue weight
+            self.wighted_queue_list.rotate(-1)
+            self.total_queue_weight = self.wighted_queue_list[self.wrr_queue_idx].weight
+
+        # Return the current selected queue
+        return current_queue.get_priority_queue()
+
     async def batch_loop(self):
         while True:
-            pri_item = await self.request_queue.get()
+            # Get the next item from the queues.
+            # Note: fetch the items given their priority,
+            # e.g., interactive requests vs background requests
+            # Note: here we wait instead of looping until there is
+            # something in the queue
+            await self.all_requests_queue.get()
+
+            # Here there is an element in one of the (weighted ordered) queues.
+            # Loop until we find that queue
+            request_queue = self.get_priority_queue()
+            while request_queue.empty():
+                # Note: get_priority_queue already returns the right queue
+                # based on the WRR algorithm
+                request_queue = self.get_priority_queue()
+
+            # The nowait should not throw since there is at least one item
+            # in the current request queue
+            pri_item = request_queue.get_nowait()
+            if pri_item is None:
+                # The current queue can be empty,
+                # e.g., for interactive jobs that are low-frequency.
+                # In this case continue until there is a queue with an item
+                continue
```

Comment:
I guess it cannot be empty because you have a while loop above and the request_queue is always not empty.
Am I correct?

Reply:
Yes, you are. This is a leftover, will remove.